src/Validator/Constraints/ExpressionValidator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Overblog\GraphQLBundle\Validator\Constraints;
+
+use Overblog\GraphQLBundle\Validator\ValidationNode;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Expression;
+use Symfony\Component\Validator\Exception\LogicException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class ExpressionValidator extends \Symfony\Component\Validator\Constraints\ExpressionValidator
+{
+    private $expressionLanguage;
+
+    public function __construct(ExpressionLanguage $expressionLanguage = null)
+    {
+        $this->expressionLanguage = $expressionLanguage;
+        parent::__construct(null, $expressionLanguage);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Expression) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Expression');
+        }
+
+        $variables = $constraint->values;
+        $variables['value'] = $value;
+
+        $object = $this->context->getObject();
+
+        $variables['this'] = $object;
+
+        if ($object instanceof ValidationNode) {
+            $variables['prevValue'] = $object->getResolverArg('value');
+            $variables['context'] = $object->getResolverArg('context');
+            $variables['args'] = $object->getResolverArg('args');
+            $variables['info'] = $object->getResolverArg('info');
+        }
+
+        if (!$this->getExpressionLanguage()->evaluate($constraint->expression, $variables)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value, self::OBJECT_TO_STRING))
+                ->setCode(Expression::EXPRESSION_FAILED_ERROR)
+                ->addViolation();
+        }
+    }
+
+    private function getExpressionLanguage()
+    {
+        if (null === $this->expressionLanguage) {
+            if (!\class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
+                throw new LogicException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed.');
+            }
+            $this->expressionLanguage = new ExpressionLanguage();
```

Comment:
why not injecting the bundle expression language service?

Reply:
This class is not declared in the bundle's `services.yml`. The declaration of this class as a service is fully on the users and is documented [here](https://github.com/murtukov/GraphQLBundle/blob/feature/validation_v.0.13/docs/validation/index.md#using-built-in-expression-functions).

This class contained this weird logic, because it's was mostly a copy-paste of the original class from the Symfony. Refactored.